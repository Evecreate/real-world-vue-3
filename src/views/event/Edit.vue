<script setup>
defineProps({
  loggedIn: {
    type: Boolean,
    required: true,
  },
});
</script>

<template>
  <p>Edit the event here</p>
  <div class="app-header">
    <button v-show="loggedIn">Edit</button>
  </div>
</template>

<style scoped>
.app-header {
  min-height: 40px;
  padding: 0.2em 1em;
  background: white;
}
</style>
