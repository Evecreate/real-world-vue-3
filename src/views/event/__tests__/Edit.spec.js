import { describe, it, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import Edit from '../Edit.vue';

describe('Edit', () => {
  it('Logged in show button', () => {
    const wrapper = mount(Edit, {
      props: {
        loggedIn: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('button').isVisible()).toBe(true);
  });
  it('Logged out do not show button', () => {
    const wrapper = mount(Edit, {
      props: {
        loggedIn: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('button').isVisible()).toBe(false);
  });
});
