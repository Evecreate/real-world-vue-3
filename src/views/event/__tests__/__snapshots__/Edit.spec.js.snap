// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Edit > Logged in show button 1`] = `
"<p data-v-9dfe3d8a="">Edit the event here</p>
<div data-v-9dfe3d8a="" class="app-header"><button data-v-9dfe3d8a="">Edit</button></div>"
`;

exports[`Edit > Logged out do not show button 1`] = `
"<p data-v-9dfe3d8a="">Edit the event here</p>
<div data-v-9dfe3d8a="" class="app-header"><button data-v-9dfe3d8a="" style="display: none;">Edit</button></div>"
`;
